sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file    ng_netreg.c
+ */
+
+#include <errno.h>
+
+#include "clist.h"
+#include "net/ng_netreg.h"
+#include "net/ng_nettype.h"
+#include "utlist.h"
+
+#define _INVALID_TYPE(type) (((type) <= NG_NETTYPE_UNDEF) || ((type) >= NG_NETTYPE_NUMOF))
+
+static ng_netreg_entry_t *netreg[NG_NETTYPE_NUMOF - 1]; /* leave out NG_NETTYPE_UNDEF */
+static ng_netreg_entry_t netreg_entries[NG_NETREG_SIZE];
+
+static inline int _netreg_entry_equal(ng_netreg_entry_t *a, ng_netreg_entry_t *b)
+{
+    /* to be pluggable with utlist, so result must be 0 when match */
+    return ((a->pid != b->pid) || (a->demux_ctx != b->demux_ctx));
+}
+
+int ng_netreg_register(ng_nettype_t type, uint16_t demux_ctx, kernel_pid_t pid)
+{
+    ng_netreg_entry_t *entry = NULL;
+
+    if (_INVALID_TYPE(type)) {
+        return -EINVAL;
+    }
+
+    for (int i = 0; i < NG_NETREG_SIZE; i++) {
```

Comment:
To me, the purpose of this loop wasn't instantly clear, maybe it would make sense to add /* look for an unclaimed entry */ or something?